shotover/src/connection.rs

```diff
@@ -412,9 +451,10 @@ impl DummyResponseInserter {
         self.pending_requests_count += requests.len();
     }
 
-    pub fn process_responses(&mut self, responses: &mut Vec<Message>) {
+    pub fn process_responses(&mut self, responses: &mut Vec<Message>, start_at: usize) {
```

Comment:
I believe this is correct, I should write some unit tests though, particularly with this extra parameter